app/services/api/webflow-api.ts

```diff
@@ -67,53 +56,66 @@ export const useSponsors = () => {
   return { data, ...rest }
 }
 
-const talksOptions = webflowOptions<RawTalk>(TALKS)
-export const useTalks = () => useQuery(talksOptions)
-
 const venuesOptions = webflowOptions<RawVenue>(VENUES)
 export const useVenues = () => useQuery(venuesOptions)
 
+const speakersOptions = webflowOptions<RawSpeaker>(SPEAKERS)
 const workshopsOptions = webflowOptions<RawWorkshop>(WORKSHOPS)
-export const useWorkshops = () => useQuery(workshopsOptions)
-
+const recurringEventsOptions = webflowOptions<RawRecurringEvents>(RECURRING_EVENTS)
+const talksOptions = webflowOptions<RawTalk>(TALKS)
 const scheduledEventsOptions = webflowOptions<RawScheduledEvent>(SCHEDULE)
-const scheduledEventsQueries = [
+
+const scheduledEventQueries = [
   speakersOptions,
   workshopsOptions,
   recurringEventsOptions,
   talksOptions,
   scheduledEventsOptions,
 ] as const
+
 export const prefetchScheduledEvents = async () => {
-  await Promise.all(
-    scheduledEventsQueries.map(async (query) => {
-      return queryClient.prefetchQuery(query)
-    }),
-  )
+  scheduledEventQueries.forEach(async (query) => {
+    await queryClient.prefetchQuery(query)
+  })
 }
-export const useScheduledEvents = () => {
+
+export const useScheduledEventsData = () => {
   const queries = useQueries({
-    queries: scheduledEventsQueries,
+    queries: scheduledEventQueries,
   })
 
-  const [speakers, workshops, recurringEvents, talks, scheduledEvents] = queries
+  const isLoading = queries.map((query) => query.isLoading).some((isLoading) => isLoading)
+  const isRefetching = queries
+    .map((query) => query.isRefetching)
+    .some((isRefetching) => isRefetching)
+  const refetch = async () => Promise.all(queries.map((query) => query.refetch()))
+
+  const [
+    { data: speakers },
+    { data: workshops },
+    { data: recurringEvents },
+    { data: talks },
+    { data: scheduledEvents },
+  ] = queries
 
   return {
-    data: cleanedSchedule({
-      recurringEvents: recurringEvents.data,
-      scheduledEvents: scheduledEvents.data,
-      speakers: cleanedSpeakers(speakers.data),
-      talks: cleanedTalks({ speakers: speakers.data, talks: talks.data }),
-      workshops: cleanedWorkshops(workshops.data, cleanedSpeakers(speakers.data)),
-    }),
-    refetch: async () => Promise.all(queries.map((query) => query.refetch())),
-    isLoading: queries.map((query) => query.isLoading).some((isLoading) => isLoading),
-    isRefetching: queries.map((query) => query.isFetching).some((isFetching) => isFetching),
+    data: isLoading
+      ? []
+      : cleanedSchedule({
+          recurringEvents,
+          scheduledEvents,
+          speakers: cleanedSpeakers(speakers),
+          talks: cleanedTalks({ speakers, talks }),
+          workshops: cleanedWorkshops(workshops, cleanedSpeakers(speakers)),
+        }),
+    isLoading,
+    isRefetching,
+    refetch,
```

Comment:
This is very much an all or nothing. If we don't have all the data, we can't display it together.